libc++ with gcc broken

## Description
Since b59552dc43c4e68f58f0ab4163b5d8f5f9e28837, builing with gcc when using libc++ as a c++ standard library (I know, I have my reasons) is broken. Previously if you had _LIBCPP_VERSION defined before including doctest.h, it picked it up and it worked (accidentally). Now, it's inside a DOCTEST_CLANG check, which means with gcc it'll try to forward declare std stuff, and it errors out if I include an stl header before doctest.

### Steps to reproduce
```
$ echo -e '#include <iostream>\n#include <doctest.h>' | gcc -x c++ - -nostdinc++ -isystem /usr/include/c++/v1 -I <path to doctest>/doctest/doctest
In file included from <stdin>:2:
doctest/doctest.h:438:8: error: ‘char_traits’ is not a class template
  438 | struct char_traits<char>;
      |        ^~~~~~~~~~~
doctest/doctest.h:438:8: error: template specifiers not specified in declaration of ‘template<class charT> struct std::char_traits’
doctest/doctest.h:441:9: error: reference to ‘basic_ostream’ is ambiguous
  441 | typedef basic_ostream<char, char_traits<char>> ostream;
      |         ^~~~~~~~~~~~~
In file included from /usr/include/c++/v1/__debug:14,
                 from /usr/include/c++/v1/utility:218,
                 from /usr/include/c++/v1/__functional_base:26,
                 from /usr/include/c++/v1/memory:808,
                 from /usr/include/c++/v1/__locale:18,
                 from /usr/include/c++/v1/ios:214,
                 from /usr/include/c++/v1/iostream:37,
                 from <stdin>:1:
/usr/include/c++/v1/iosfwd:130:32: note: candidates are: ‘template<class _CharT, class _Traits> class std::__1::basic_ostream’
  130 |     class _LIBCPP_TEMPLATE_VIS basic_ostream;
      |                                ^~~~~~~~~~~~~
In file included from <stdin>:2:
doctest/doctest.h:440:7: note:                 ‘template<class charT, class traits> class std::basic_ostream’
  440 | class basic_ostream;
      | 
```

### Extra information
* doctest version: **v2.4.9**
* Operating System: **Gentoo linux**
* Compiler+version: **gcc 8.2.0 arm, 11.3.0 amd64, 12.1.1 amd64**

--- a/doctest/doctest.h
+++ b/doctest/doctest.h
@@ -484,12 +484,13 @@ DOCTEST_GCC_SUPPRESS_WARNING_POP
 // https://github.com/doctest/doctest/issues/356
 #if DOCTEST_CLANG
 #include <ciso646>
+#endif // clang
+
 #ifdef _LIBCPP_VERSION
 #ifndef DOCTEST_CONFIG_USE_STD_HEADERS
 #define DOCTEST_CONFIG_USE_STD_HEADERS
 #endif
 #endif // _LIBCPP_VERSION
-#endif // clang
 
 #ifdef DOCTEST_CONFIG_USE_STD_HEADERS
 #ifndef DOCTEST_CONFIG_INCLUDE_TYPE_TRAITS

--- a/doctest/parts/doctest_fwd.h
+++ b/doctest/parts/doctest_fwd.h
@@ -481,12 +481,13 @@ DOCTEST_GCC_SUPPRESS_WARNING_POP
 // https://github.com/doctest/doctest/issues/356
 #if DOCTEST_CLANG
 #include <ciso646>
+#endif // clang
+
 #ifdef _LIBCPP_VERSION
 #ifndef DOCTEST_CONFIG_USE_STD_HEADERS
 #define DOCTEST_CONFIG_USE_STD_HEADERS
 #endif
 #endif // _LIBCPP_VERSION
-#endif // clang
 
 #ifdef DOCTEST_CONFIG_USE_STD_HEADERS
 #ifndef DOCTEST_CONFIG_INCLUDE_TYPE_TRAITS

